File: Boca/JAVA_2/Q9/src/Livro.java
public class Livro extends Produto {
    private String autor;

    public Livro (String nome, double preco, String autor) {
        super(nome, preco);
        this.autor = autor;
    }

    public String getAutor() {
        return autor;
    }

    @Override
    public String toString() {
        return super.toString() + String.format(" - %s", autor);
    }
    // Harry Potter e as Reliquias da Morte (R$ 39.51) - J. K. Rowling
}

File: Boca/CPP/06/Livro.cpp
#include "Livro.hpp" 

Livro::Livro(const string& nome, double preco, const string& autor) : Produto(nome, preco) {
    this->autor = autor;
}

void Livro::print(ostream &out) const {
    Produto::print(out);
    out << " - " << autor;
}

ostream& operator<< (ostream &out, Livro &livro) {
    livro.print(out);
    return out;
}

Livro::~Livro() {}